### PR TITLE
Fix a panic on node.Deregister fail

### PR DESCRIPTION
**Pull request by notnoop**
And remove an unused field.

### CODE REVIEW

File: nomad/core_sched.go [hunk 1]
```diff
@@ -494,7 +494,7 @@ OUTER:
 		}
 		var resp structs.NodeUpdateResponse
 		if err := c.srv.RPC("Node.Deregister", &req, &resp); err != nil {
-			c.logger.Error("node reap failed", "node_id", nodeID, err)
+			c.logger.Error("node reap failed", "node_id", nodeID, "error", err)
```
Comment (notnoop) [inline nomad/core_sched.go]: Is there a quick way to ensure that logging functions get an odd number of arguments?  Maybe a linter?

Comment (preetapan) [reply nomad/core_sched.go]: thanks to @evanphx, we do now! https://github.com/hashicorp/go-hclog/pull/31 (still need to add it to our make check)

Comment (notnoop) [reply nomad/core_sched.go]: Yeah - it came in handy - with it, I found three more cases of such error in 7840ab51e2ca74f7ae5cd38933d1419900b72532!

Comment (notnoop) [reply nomad/core_sched.go]: Also, planning to add it to our linters after https://github.com/hashicorp/go-hclog/pull/32 is merged.

**Pull request merged**